packages/react/src/components/select/dataUpdater.ts

```diff
@@ -10,58 +16,125 @@ import {
+const dataUpdater = (
+  event: ChangeEvent,
+  dataHandlers: SelectDataHandlers,
+): { didSearchChange: boolean; didSelectionsChange: boolean; didDataChange: boolean } => {
   const { id, type, payload } = event as ChangeEvent<EventId, EventType>;
   const current = dataHandlers.getData();
+  const returnValue = {
+    didSearchChange: false,
+    didSelectionsChange: false,
+    didDataChange: false,
+  };
+  if (current.disabled) {
+    return returnValue;
+  }
   const openOrClose = (open: boolean) => {
+    if (current.open === open) {
+      return false;
+    }
     const now = Date.now();
-    if (
-      dataHandlers.getMetaData().lastToggleCommand &&
-      now - dataHandlers.getMetaData().lastToggleCommand < MIN_USER_INTERACTION_TIME_IN_MS
-    ) {
+    if (now - dataHandlers.getMetaData().lastToggleCommand < MIN_USER_INTERACTION_TIME_IN_MS) {
       return false;
     }
     dataHandlers.updateData({ open });
     dataHandlers.updateMetaData({ lastToggleCommand: now });
     return true;
   };
 
-  const updateGroups = (groups: SelectData['groups']) => {
+  const updateGroups = (groups: SelectData['groups'], clickedOption?: Option) => {
     dataHandlers.updateData({ groups });
+    dataHandlers.updateMetaData({
+      selectedOptions: createSelectedOptionsList(dataHandlers.getMetaData().selectedOptions, groups),
+      lastClickedOption: clickedOption,
+      // textContent is re-created, when a textProvider is called
+      textContent: undefined,
+    });
   };
 
   if (isOpenOrCloseEvent(id, type)) {
     const willOpen = !current.open;
-    return openOrClose(willOpen);
+    const didUpdate = openOrClose(willOpen);
+    return {
+      ...returnValue,
+      didDataChange: didUpdate,
+    };
   }
 
   if (isOptionClickEvent(id, type)) {
     const clickedOption = payload && (payload.value as Option);
     if (!clickedOption) {
-      return false;
+      return returnValue;
     }
     const newGroups = updateSelectedOptionInGroups(current.groups, {
       ...clickedOption,
       selected: !clickedOption.selected,
     });
-    updateGroups(newGroups);
+    updateGroups(newGroups, clickedOption);
     openOrClose(false);
-
-    return true;
+    return {
+      ...returnValue,
+      didSelectionsChange: true,
+      didDataChange: true,
+    };
   }
 
   if (isClearOptionsClickEvent(id, type)) {
     const newGroups = clearAllSelectedOptions(current.groups);
     updateGroups(newGroups);
-    return true;
+    return {
+      ...returnValue,
+      didSelectionsChange: true,
+      didDataChange: true,
+    };
   }
 
   if (isOutsideClickEvent(id, type) || isCloseEvent(id, type)) {
-    return openOrClose(false);
+    if (openOrClose(false)) {
+      return {
+        ...returnValue,
+        didDataChange: true,
+      };
+    }
+  }
+
+  return returnValue;
+};
+
+export const changeChandler: ChangeHandler<SelectData, SelectMetaData> = (event, dataHandlers): boolean => {
```

Comment:
just out of curiosity, why is it change**C**handler and not changeHandler?

Reply:
Oh, how blind you can be to your own code. 😄 That misspelling has been there all along. Good catch! Changed to "changeHandler"